kdf.c

```diff
@@ -109,8 +89,12 @@ void kdf_decryption_key(const char *username, const char *password, int iteratio
 		iterations = 1;
 
 	if (iterations == 1)
-		sha256_hash(user_lower, strlen(user_lower), password, strlen(password), hash);
+		fail_invalid_iteration_count();
 	else
 		pbkdf2_hash(user_lower, strlen(user_lower), password, strlen(password), iterations, hash);
 	mlock(hash, KDF_HASH_LEN);
 }
+
+void fail_invalid_iteration_count() {
```

Comment:
This function is small, so to prevent the overhead of calling such a small function consider marking it as `inline`. Also consider marking it as `static` because this is only meant to be used within the scope of `kdf.c`. Example:

`static inline void fail_invalid_iteration_count()`